[Khoo Jing Hong, Derrick] iP

drkkjh opened the pull request:
# Duke

>"Simplicity is the ultimate sophistication." ~ [Leonardo Da Vinci](https://sourcesofinsight.com/simplicity-quotes/#:~:text=Here%20are%20my%20top%20three,with%20simplicity.%E2%80%9D%20%E2%80%94%20Isaac%20Newton)

## The CLI chatbot that helps you keep track of your tasks. It's that simple.

### Features include:
- text-based UI
- ~~super~~ **EXTREMELY** fast response

### Type of tasks duke can track:
1. Todos
2. Deadlines
3. Events

### Sample input:
```java
deadline return book /by 2022-08-30
```

### Features:
- [x] Manage your tasks 😀 
- [ ] \(Maybe in the future) Create *reminders* 🤡 

`inline code`

Commit:
Add Gradle support

drkkjh committed:
added Level-1

drkkjh committed:
added Level-2: Add, List

drkkjh committed:
added Level-3: Mark as Done

drkkjh committed:
added Level-4: ToDos, Events, Deadlines

drkkjh committed:
added A-TextUITesting

drkkjh committed:
added Level-5: Handle errors and updated automated testing inputs and outputs

drkkjh committed:
added Level-6: Delete and edited expected and input files

drkkjh committed:
added A-ENUMS and modified appropriate methods to be static in Duke.java

drkkjh committed:
Add Level-7: Save feature

drkkjh committed:
Add Level-8: Dates and Times

drkkjh committed:
Merge branch 'branch-Level-7'

drkkjh committed:
Merge branch 'branch-Level-8'

drkkjh committed:
Add new class TaskList.java to store Task objects

drkkjh committed:
Add new class Storage to save/load to hard disk

drkkjh committed:
Add A-MoreOOP: AddCommand class

drkkjh committed:
Add A-MoreOOP: Command class

drkkjh committed:
Add A-MoreOOP: Command class

drkkjh committed:
Add A-MoreOOP: DeleteCommand class

drkkjh committed:
Edit Duke.java for A-MoreOOP

drkkjh committed:
Add A-MoreOOP: ListCommand class

drkkjh committed:
Add A-MoreOOP: MarkDoneCommand class

drkkjh committed:
Add A-MoreOOP: OnDateCommand class

drkkjh committed:
Add A-OOP: OnDateCommand class

drkkjh committed:
Add A-MoreOOP: Parser class

drkkjh committed:
Edit Storage class to accomodate taskList class

drkkjh committed:
Edit TaskList class with getters and to accomodate Command

drkkjh committed:
Add A-MoreOOP: Ui class

drkkjh committed:
Update .gitignore

drkkjh committed:
Add packages to code: A-Packages

drkkjh committed:
Merge branch 'add-gradle-support'
Add gradle support

drkkjh committed:
Rename mainClassName in build.gradle

drkkjh committed:
Add J-Unit tests

drkkjh committed:
Add JavaDocs to all public classes and methods

drkkjh committed:
Rewrite code that violates Coding Standard

drkkjh committed:
Add Level-9: Find

drkkjh committed:
Merge branch 'branch-A-JavaDoc'

drkkjh committed:
Merge branch 'branch-A-CodingStandard'

drkkjh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/task/Task.java

drkkjh committed:
Refactor code due to violation of coding standards

drkkjh committed:
Refactor code to throw exception when incorrect format entered for date

drkkjh committed:
Add A-Checkstyle and refactor code that has coding violations

drkkjh committed:
Merge branch 'A-CheckStyle'

donljh commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,39 @@
+        return true;
+    }
+
+    /**
+     * Handles the command to exit Duke.
+     * @param storage  the <code>Storage</code> dealing with loading and saving tasks in a file
+     * @param ui       the user interfaces that deals with user inputs
+     * @param taskList the list of tasks
+     * @throws DukeException if there is an error saving to the file
+     */
+
```
Perhaps a line between the method description and parameters would better follow the coding standards here? I believe this could apply to other similar parts of your code.

Referencing the comments section in - https://se-education.org/guides/conventions/java/intermediate.html

donljh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,81 @@
+     * @return true if and only if the queried date is the same as the event-type
+     *      task's event date.
+     */
+    public boolean isHappeningOnDate(LocalDate localDate) {
+        return this.at.equals(localDate);
+    }
```
I like that you used the appropriate boolean naming standard here.

donljh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+            second = strArray[1];
+        }
+        Command cmd;
+        switch (first) {
+        case "bye":
+            cmd = new ByeCommand();
+            break;
+        case "list":
+            cmd = new ListCommand();
+            break;
+        case "mark":
+            cmd = new MarkDoneCommand(parseInt(second));
+            break;
+        case "unmark":
+            cmd = new MarkUndoneCommand(parseInt(second));
+            break;
+        case "todo":
+            cmd = new AddCommand(handleTodo(second));
+            break;
+        case "deadline":
+            cmd = new AddCommand(handleDeadline(second));
+            break;
+        case "event":
+            cmd = new AddCommand(handleEvent(second));
+            break;
+        case "delete":
+            cmd = new DeleteCommand(parseInt(second));
+            break;
+        case "on":
+            cmd = new OnDateCommand(LocalDate.parse(second));
+            break;
+        case "find":
+            cmd = new FindCommand(handleFind(second));
+            break;
+        default:
```
I like that you followed the appropriate indentation rules for switch clause here.

donljh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+    /**
+     * Constructs an instance of Duke.
+     *
+     * @param filePath the file location for saving data
```
Maybe you could consider adding punctuations behind parameter descriptions to better comply with the coding standards? I think this could apply to other parts of your code too.

Referencing the comments section in - https://se-education.org/guides/conventions/java/intermediate.html

donljh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+
+import java.time.LocalDate;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkDoneCommand;
+import duke.command.MarkUndoneCommand;
+import duke.command.OnDateCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
```
I like that you explicitly imported each class instead of a wildcard import here.

drkkjh committed:
Refactor code to add GUI to Duke

drkkjh committed:
Merge branch 'branch-Level-10'

zylee348 commented on src/main/java/duke/command/Commands.java:
```diff
@@ -0,0 +1,8 @@
+/**
+ * Enums for duke
+ */
+public enum Commands {
+    TODO, DEADLINE, EVENT
```
good to see proper use of enums

drkkjh committed:
Update build.gradle to support javaFX

drkkjh committed:
# Add assertions

### Currently there are no assertions present in code.

### This decreases the readability of the code where important
assumptions are made about in various segments in the code, for
important functionalities.

### Let's add assertions to the following classes:
* `Parser`
* `Event`
* `Deadline`
* `Todo`
* `Task`
* `TaskList`

drkkjh committed:
Improve code quality: A-CodeQuality

Code duplication and magic numbers reduces code readability and also,
code quality.

The multiple error messages of the same message causes code duplication.
Magic numbers that exist when accessing array through indexing
causes a decrease in code quality.

Let's reduce code duplication and usage of magic numbers in
the following classes:
* `MainWindow`
* `Storage`
* `Deadline`
* `Event`
* `Todo`
* `Parser`

drkkjh committed:
Add usage of streams: A-Streams

drkkjh committed:
Merge pull request #1 from drkkjh/branch-A-Assertions

Add assertions to Duke

drkkjh committed:
Merge branch 'master' into branch-A-Assertions

* master:

drkkjh committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  # Add assertions

# Conflicts:
#	src/main/java/duke/task/Todo.java

drkkjh committed:
Merge branch 'master' into branch-A-Streams

* master:
  # Add assertions

drkkjh committed:
Merge pull request #2 from drkkjh/branch-A-CodeQuality

Branch a code quality

drkkjh committed:
Merge branch 'master' into branch-A-CodeQuality

* master:

drkkjh committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Improve code quality: A-CodeQuality

drkkjh committed:
Merge branch 'master' into branch-A-Streams

* master:
  Improve code quality: A-CodeQuality

drkkjh committed:
Merge pull request #3 from drkkjh/branch-A-Streams

Branch a streams

drkkjh committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Add usage of streams: A-Streams

drkkjh committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add usage of streams: A-Streams

drkkjh committed:
Merge branch 'master' into branch-A-Streams

* master:

drkkjh committed:
Add CI support: A-CI

drkkjh committed:
Fix minor bug of unable to parse input

drkkjh committed:
Add ability for user to tag tasks: C-Tagging

drkkjh committed:
Merge branch 'branch-C-Tagging'

* branch-C-Tagging:
  Add ability for user to tag tasks: C-Tagging

drkkjh committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:

drkkjh committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

drkkjh committed:
Merge branch 'branch-A-Streams'

* branch-A-Streams:

drkkjh committed:
Update workflow

drkkjh committed:
Add personality: Change name to Zeus

drkkjh committed:
Add circular profile picture and lengthen dialogbox: A-BetterGui

drkkjh committed:
Add untag command as extension to C-Tagging

drkkjh committed:
Add more erroneous handling: A-MoreErrorHandling

drkkjh committed:
Fix GUI issue of user being able to resize window

drkkjh committed:
Add credit given to author for code reuse

drkkjh committed:
Remove unused pictures

drkkjh committed:
Fix bugs related to saving data and formatting of tags

drkkjh committed:
Add User Guide for Zeus

drkkjh committed:
Add link to download jar file for Zeus in readme
